src/vi/doctruyen3q/src/eu/kanade/tachiyomi/extension/vi/doctruyen3q/DocTruyen3Q.kt

```diff
@@ -0,0 +1,85 @@
+package eu.kanade.tachiyomi.extension.vi.doctruyen3q
+
+import eu.kanade.tachiyomi.multisrc.wpcomics.WPComics
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class DocTruyen3Q : WPComics(
+    "DocTruyen3Q",
+    "https://doctruyen3qk.pro",
+    "vi",
+    dateFormat = SimpleDateFormat("dd-MM-yyyy", Locale.getDefault()),
+    gmtOffset = null,
+) {
+    override val client = super.client.newBuilder()
+        .rateLimit(3)
+        .build()
+
+    override fun pageListParse(document: Document): List<Page> {
+        return document.select(".page-chapter a img, .page-chapter img").mapIndexed { index, element ->
+            val img = element.attr("abs:src").takeIf { it.isNotBlank() } ?: element.attr("abs:data-original")
+            Page(index, imageUrl = img)
+        }.distinctBy { it.imageUrl }
+    }
+
+    override fun popularMangaSelector() = "div.item-manga div.item"
+
+    override fun popularMangaFromElement(element: Element) = SManga.create().apply {
+        element.select("h3 a").let {
+            title = it.text()
+            setUrlWithoutDomain(it.attr("abs:href"))
+        }
+        thumbnail_url = element.selectFirst("img")?.attr("abs:src")
+    }
+
+    override fun searchMangaSelector() = popularMangaSelector()
+
+    override fun searchMangaFromElement(element: Element) = popularMangaFromElement(element)
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$baseUrl/$searchPath".toHttpUrl().newBuilder()
+
+        filters.forEach { filter ->
+            when (filter) {
+                is GenreFilter -> filter.toUriPart()?.let { url.addPathSegment(it) }
+                is StatusFilter -> filter.toUriPart()?.let { url.addQueryParameter("status", it) }
+                else -> {}
+            }
+        }
+
+        when {
+            query.isNotBlank() -> url.addQueryParameter(queryParam, query)
+            else -> url.addQueryParameter("page", page.toString())
```

Comment:
Is this correct? It might cause issues with duplicate entries since it will only request page 1 when query is set.

Reply:
yes it correct